src/material/button/_button-base.scss

```diff
@@ -81,6 +81,8 @@
     @include mdc-touch-target.touch-target(
       $set-width: $is-square,
       $query: mdc-helpers.$mdc-base-styles-query);
+
+    pointer-events: none;
```

Comment:
This change defeats the purpose of the touch target. The point of it is that it should be clickable so that the button has a larger touch area.

Reply:
I changed the logic, instead of `pointer-event` I add the `z-index` for the mat-icon inside the button.